### PR TITLE
Upgrade hadolint to version 2.3.0

### LINKED ISSUE #15544: Bump hadolint to v2.3.0
We are currently using some old version of Hadolint. 
https://github.com/apache/airflow/blob/4b031d39e12110f337151cda6693e2541bf71c2c/scripts/ci/static_checks/lint_dockerfile.sh#L22
I think it's worth updating to better validate our Docker files. It shouldn't be very time-consuming, but we'll have to fix some warnings. I am most interested in this rule that will optimize our images better.
> DL3059: Multiple consecutive RUN instructions (https://github.com/hadolint/hadolint/pull/591)

### PR DESCRIPTION
**Pull request by potiuk**
Newer versions of hadolint hint about more Docker problems:

* consecutive RUN operation
* invalid labels

This PR fixes all the problems reported in our dockerfiles
by the latest hadolint and refreshes all our images used in CI
and chart so that corrected label names are included (one of
the errors in all our dockerfiles turned out to be camel-case
and - in label keys, which is not valid according to
Docker label key specification.

Fixes: #15544

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: chart/dockerfiles/pgbouncer/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # under the License.
 ARG ALPINE_VERSION="3.12"
 FROM alpine:${ALPINE_VERSION} AS builder
-SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
+SHELL ["/bin/ash", "-e", "-x", "-c", "-o", "pipefail"]
```
Comment (mik-laj) [inline chart/dockerfiles/pgbouncer/Dockerfile:19]: Ash?

Comment (potiuk) [reply chart/dockerfiles/pgbouncer/Dockerfile:19]: Yep. https://en.wikipedia.org/wiki/Almquist_shell provided by busybox (in alpine image). Alpine has no bash. But hadolint suggest to use `ash` instead.

**Review by dimberman — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -191,9 +191,8 @@ ENV INSTALL_MYSQL_CLIENT=${INSTALL_MYSQL_CLIENT} \
 
 # Only copy install_mysql.sh to not invalidate cache on other script changes
 COPY scripts/docker/install_mysql.sh /scripts/docker/install_mysql.sh
-RUN bash ./scripts/docker/install_mysql.sh dev
-
-RUN mkdir -p /root/.local/bin
+RUN bash ./scripts/docker/install_mysql.sh dev && \
+    mkdir -p /root/.local/bin
```
Comment (mik-laj) [inline Dockerfile]: Is it related to Python or MySQL? If Python, then we can move that to the 204 line.

File: Dockerfile.ci [hunk 3]
```diff
@@ -97,9 +97,8 @@ RUN mkdir -pv /usr/share/man/man1 \
 
 # Only copy install_mysql.sh to not invalidate cache on other script changes
 COPY scripts/docker/install_mysql.sh /scripts/docker/install_mysql.sh
-RUN bash /scripts/docker/install_mysql.sh dev
-
-RUN adduser airflow \
+RUN bash /scripts/docker/install_mysql.sh dev && \
+    adduser airflow \
```
Comment (mik-laj) [inline Dockerfile.ci]:
```suggestion
RUN bash /scripts/docker/install_mysql.sh dev \
    && adduser airflow \
```

File: chart/dockerfiles/pgbouncer/Dockerfile [hunk 4]
```diff
@@ -30,15 +30,16 @@ RUN apk --no-cache add make pkgconfig build-base libtool wget gcc g++ libevent-d
 # We are not using Dash so we can safely ignore the "Dash warning"
 # "In dash, something is not supported." https://github.com/koalaman/shellcheck/wiki/SC2169
 # hadolint ignore=SC2169
-RUN wget "https://github.com/pgbouncer/pgbouncer/releases/download/pgbouncer_${PGBOUNCER_VERSION//\./_}/pgbouncer-${PGBOUNCER_VERSION}.tar.gz"
-RUN echo "${PGBOUNCER_SHA256}  pgbouncer-${PGBOUNCER_VERSION}.tar.gz" | sha256sum -c -
-RUN tar -xzvf pgbouncer-$PGBOUNCER_VERSION.tar.gz
+RUN wget --progress=dot:giga "https://github.com/pgbouncer/pgbouncer/releases/download/pgbouncer_${PGBOUNCER_VERSION//\./_}/pgbouncer-${PGBOUNCER_VERSION}.tar.gz" \
+  && echo "${PGBOUNCER_SHA256}  pgbouncer-${PGBOUNCER_VERSION}.tar.gz" | sha256sum -c - \
+  && tar -xzvf pgbouncer-$PGBOUNCER_VERSION.tar.gz
```
Comment (mik-laj) [inline chart/dockerfiles/pgbouncer/Dockerfile]:
```suggestion
    && echo "${PGBOUNCER_SHA256}  pgbouncer-${PGBOUNCER_VERSION}.tar.gz" | sha256sum -c - \
    && tar -xzvf pgbouncer-$PGBOUNCER_VERSION.tar.gz
```

File: chart/dockerfiles/pgbouncer/Dockerfile [hunk 5]
```diff
@@ -30,15 +30,16 @@ RUN apk --no-cache add make pkgconfig build-base libtool wget gcc g++ libevent-d
...
-RUN ./configure --prefix=/usr --disable-debug && make && make install
-RUN mkdir /etc/pgbouncer
-RUN cp ./etc/pgbouncer.ini /etc/pgbouncer/
-RUN touch /etc/pgbouncer/userlist.txt
-RUN sed -i -e "s|logfile = |#logfile = |"  \
+RUN ./configure --prefix=/usr --disable-debug && make && make install \
+ && mkdir /etc/pgbouncer \
+ && cp ./etc/pgbouncer.ini /etc/pgbouncer/ \
+ && touch /etc/pgbouncer/userlist.txt \
+ && sed -i -e "s|logfile = |#logfile = |"  \
```
Comment (mik-laj) [inline chart/dockerfiles/pgbouncer/Dockerfile]:
```suggestion
    && mkdir /etc/pgbouncer \
    && cp ./etc/pgbouncer.ini /etc/pgbouncer/ \
    && touch /etc/pgbouncer/userlist.txt \
    && sed -i -e "s|logfile = |#logfile = |"  \
```

**Review by mik-laj — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (potiuk) [reply Dockerfile]: Yep. might be better there.

**Pull request merged**